src/components-examples/material/table/table-reorderable/table-reorderable-example.ts

```diff
@@ -0,0 +1,46 @@
+import { Component } from '@angular/core';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+
+/**
+ * @title Table with re-orderable columns
+ */
+@Component({
+  selector: 'table-reorderable-example',
+  templateUrl: './table-reorderable-example.html',
+  styleUrls: [ './table-reorderable-example.css' ]
+})
+export class TableReorderableExample {
+  private originalColumnPositions: string[] = ['position', 'name', 'weight', 'symbol'];
+  columns: string[] = Object.assign([], this.originalColumnPositions);
+  dataSource = ELEMENT_DATA;
+
+  drop(event: CdkDragDrop<string[]>) {
+    moveItemInArray(this.columns, this.getCurPosition(event.previousIndex), event.currentIndex);
+  }
+
+  getCurPosition(colOldIndex: number): number {
```

Comment:
This definitely shouldn't be necessary and we probably shouldn't propagate it in the example. Were you seeing this issue before version 9?

Reply:
It did seem like a bug to me. However, I wasn't able to reproduce it in stackblitz because the version of angular in stackblitz doesn't have fix to https://github.com/angular/components/issues/15948, so I can't get far enough into the drag-drop usecase to showcase this maybe-a-bug.

Would you like me to file a bug anyway?

Reply:
FYI, here is the note from my original PR description:

Note: It looks like CdkDragDrop.previousIndex is never being updated. It always reports the original index - rather than the previous - of the thing being dragged. So, in this PR I've had to add a little mapping from CdkDragDrop.previousIndex -> name from column list -> actual previousIndex.

Reply:
Yes, it would be helpful to have it captured in a bug report. We have #15948 in master so I can try it out against the dev app.

Reply:
https://github.com/angular/components/issues/18482

Reply:
Done.